src/sc/sidechain.cpp

```diff
@@ -757,12 +760,12 @@ void CSidechain::InitScFees()
         if (!isNonCeasing()) {
             Sidechain::ScFeeData defaultData(
                 lastTopQualityCertView.forwardTransferScFee, lastTopQualityCertView.mainchainBackwardTransferRequestScFee);
-            scFees.push_back(defaultData);
+            scFees.emplace_back(defaultData);
```

Comment:
Hmmm ... sorry that's not what I meant.
By using emplace_back in this way your constructing a temporary object and then in the container is copy constructed the final object.
It should be

`scFees.emplace_back(lastTopQualityCertView.forwardTransferScFee, lastTopQualityCertView.mainchainBackwardTransferRequestScFee)
`
This forwards constructor arguments to the container which will construct object in place

Reply:
My bad. 2522a6032b25bc34777cfd408f489058fb189cee and d39a7f376cd683ff02304198fef02b53898b8375 improved the object emplacement in the lists.